bluemira/equilibria/optimisation/harmonics/toroidal_harmonics_approx_functions.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-FileCopyrightText: 2021-present M. Coleman, J. Cook, F. Franza
+# SPDX-FileCopyrightText: 2021-present I.A. Maione, S. McIntosh
+# SPDX-FileCopyrightText: 2021-present J. Morris, D. Short
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+"""
+A collection of functions used to approximate toroidal harmonics.
+"""
+
+from math import factorial
+
+import numpy as np
+from scipy.special import gamma, poch
+
+
+def f_hypergeometric(a, b, c, z, n_max=20):
+    """Evaluates the hypergeometric power series up to n_max.
+    Valid for |z| < 1
+    .. math::
+        F(a, b; c; z) = \\sum_0^{n_max} \\frac{(a)_{s} (b)_{s}}{Gamma(c + s) s!} z^{s}
+
+    See https://dlmf.nist.gov/15.2#E2 and https://dlmf.nist.gov/5.2#iii for more
+    information.
+
+    """
+    F = 0
+    for s in range(n_max + 1):
+        F += (poch(a, s) * poch(b, s)) / (gamma(c + s) * factorial(s)) * z**s
+    return F
+
+
+def my_legendre_p(lam, mu, x, n_max=20):
```

Comment:
Could we remove the "my_" from the function names now we're no longer testing? Here, any use and the 'my_legrende_q' below
```suggestion
def legendre_p(lam, mu, x, n_max=20):
```